### ISSUE TITLE
script_execute_ext : argument 2 is not an array

### ISSUE THREAD

**Issue by VeroSquirrel** [zbanack/shampoo#8]
After the latest update in GameMaker Studio 2, the framework no longer functions properly, with the clean download of the latest version, we get the following error. Same goes for our currently in-development game.

```

___________________________________________
############################################################################################
ERROR in
action number 1
of Draw Event
for object obj_shampoo:

script_execute_ext : argument 2 is not an array
 at gml_Script_anon_ShCanvas_gml_GlobalScript_ShampooCanvas_11584_ShCanvas_gml_GlobalScript_ShampooCanvas (line 547) -                      script_execute_ext(meta_onDraw[0], meta_onDraw[1]);       
############################################################################################
gml_Script_anon_ShCanvas_gml_GlobalScript_ShampooCanvas_11584_ShCanvas_gml_GlobalScript_ShampooCanvas (line 547)
gml_Script_anon_Shampoo_gml_GlobalScript_ShampooMain_7907_Shampoo_gml_GlobalScript_ShampooMain (line 304) -                      canvas.render();
gml_Script_anon_Shampoo_gml_GlobalScript_ShampooMain_12959_Shampoo_gml_GlobalScript_ShampooMain (line 480) -                      render_canvas(canvases_list[| i], true);
gml_Script_shampoo_render (line 36) -        SHAMPOO.render_canvases();
gml_Object_obj_shampoo_Draw_64 (line 1) - shampoo_render();
```

**Comment by VeroSquirrel** [comment]
A quick somewhat funny mention: I tried getting around this by adding the "is_array()" function just before that specific call and while the is_array returns true, the call still fails for the same reason seen above.

**Comment by zbanack** [comment]
Thanks for the notice, I've fixed it in 0.9.4 [](https://zackbanack.itch.io/shampoo/devlog/351252/v094-minor-release)

It looks like script_execute_ext* doesn't like when empty arrays are passed into the second argument.
